sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -270,11 +260,141 @@ class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
+                            "Refresh triggered for key: %s label: %s",
+                            registration.key_filter,
+                            registration.label_filter,
+                        )
+                        updated_keys = True
+            if updated_keys:
+                self._dict = updated_dict
+                # pylint:disable=protected-access
+                self._configuration_refresh.refresh_options._refresh_registrations = updated_registrations
+                self._configuration_refresh.updated_configurations()
+        except Exception as error:  # pylint:disable=broad-except
+            # refresh should never throw an exception
+            self._configuration_refresh.failed_update(
+                error,
+                "An error occurred while checking for configuration updates. \
+                %s attempts have been made.\n %r",
+            )
+
+    def _load_all(self, **kwargs):
+        configuration_settings = {}
+        for select in self._selects:
+            configurations = self._client.list_configuration_settings(
+                key_filter=select.key_filter, label_filter=select.label_filter, **kwargs
+            )
+            for config in configurations:
+                key = self._proccess_key_name(config)
+                value = self._proccess_key_value(config)
+
+                if isinstance(config, FeatureFlagConfigurationSetting):
+                    feature_management = self._dict.get(FEATURE_MANAGEMENT_KEY, {})
+                    feature_management[key] = value
+                    if FEATURE_MANAGEMENT_KEY not in self.keys():
+                        configuration_settings[FEATURE_MANAGEMENT_KEY] = feature_management
+                else:
+                    configuration_settings[key] = value
+        self._dict = configuration_settings
+
+    def _proccess_key_name(self, config):
+        trimmed_key = config.key
+        # Trim the key if it starts with one of the prefixes provided
+        for trim in self._trim_prefixes:
+            if config.key.startswith(trim):
+                trimmed_key = config.key[len(trim) :]
+                break
+        if isinstance(config, FeatureFlagConfigurationSetting) and trimmed_key.startswith(FEATURE_FLAG_PREFIX):
+            return trimmed_key[len(FEATURE_FLAG_PREFIX) :]
+        return trimmed_key
+
+    def _proccess_key_value(self, config):
+        if isinstance(config, SecretReferenceConfigurationSetting):
+            return _resolve_keyvault_reference(config, self)
+        if _is_json_content_type(config.content_type) and not isinstance(config, FeatureFlagConfigurationSetting):
+            # Feature flags are of type json, but don't treat them as such
+            try:
+                return json.loads(config.value)
+            except json.JSONDecodeError:
+                # If the value is not a valid JSON, treat it like regular string value
+                return config.value
+        return config.value
```

Comment:
Is there no flattening for feature flag value yet?

Reply:
No. Feature Flags will be in a separate PR.